%\VignetteIndexEntry{ComSeq}
\documentclass[11pt]{article}

\usepackage{amsmath,epsfig,fullpage,hyperref}
\usepackage{underscore}
\parindent 0in

%\newcommand{\Robject}[1]{{\texttt{#1}}}
%\newcommand{\Rfunction}[1]{{\texttt{#1}}}
%\newcommand{\Rpackage}[1]{{\textit{#1}}}

\begin{document}
\SweaveOpts{concordance=TRUE}

\title{\bf Introduction to ComSeq package}

\author{Dongmei Li}

\maketitle

\begin{center}
Clinical and Translational Science Institute, University of Rochester School of Medicine and Dentistry, Rochester, NY 14642-0708\\
\end{center}

\tableofcontents

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Overview}

This document provides an introduction to the {\tt ComSeq} package. The {\tt ComSeq} package use results from four current methods (DESeq2, 
EBSeq, SAMSeq, NOISeq) for RNA-Seq differential analysis and declare a gene is significant if the gene is detected by all four methods. The rank of the 
top candidate gene is based on a weighted algorithm that put more weights on the results from the DESeq2 and EBSeq methods than results 
from the SAMSeq and NOISeq method.  The purpose of the combine package is to reduce the false discovery rate in gene differential analysis 
for RNA-Seq experiments with small sample sizes. Simulation studies have shown the combine package could control the FDR within the 
normial level of 5\% even when the sample size is as small as three in each group.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Getting started}

The {\tt ComSeq} package can be installed and loaded through the following R code.

Install the {\tt ComSeq} package with:
%code 1
<<eval=TRUE, echo=TRUE>>=
source("http://bioconductor.org/biocLite.R")
biocLite("ComSeq")
@

Load the {\tt ComSeq} package with:

%code 2
<<eval=TRUE, echo=TRUE>>=
library(ComSeq)
@

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{{\tt Combined_method} functions}

The {\tt Combined_method} function in the {\tt ComSeq} package requires input data in the matrix format with rows denoting features and 
columns denoting samples. {\tt Combined_method} is used for two-group comparisons such as study designs with a treatment group and a 
control group, which is the most common contrast in biological and biomedical field. To simplify the use of the function by biological and
biomedical users, the {\tt Combined_method} function takes many default settings in DESeq2, EBSeq, SAMSeq, NOISeq functions for two-group
comparisons and only take three arguments as input: RNA-Seq count matrix, group label, and alpha value for false discovery rate control.
The RNA-Seq data was first filtered out genes with summarized count across all samples less than 10, then normalized using the TMM method. 
The TMM normalization method has relatively better performance than other methods in terms of smaller coefficient of variation shown by 
previous literature. For group label, the {\tt Combined_method} function needs a vector, i.e., "1" denotes the treatment group and "0" 
denotes the control group. The default value for false discovery rate is 5\%.

The gene expression in C57BL/6J and DBA/2J mouse striatum example is used to illustrate the application of {\tt Comined_method} in 
identifying differentially expressed genes from RNA-Seq data. The C57BL/6J (B6) and DBA/2J (D2) mouse strains are commonly used in 
neuroscience research. The RNA-Seq data includes 10 B6 mouse strains and 11 D2 strains with an average of 22 million short sequencing 
reads generated per sample. The reads were aligned to the mouse reference genome, and counts from 16,183 Ensembl genes for both strains 
were generated after preprocessing. Genes with summarized count across all 21 samples less than 10 were first filtered out from the study. 
Then the TMM method was used to normalize the RNA-Seq count data before the differential analysis. The normalized RNA-Seq data were 
analyzed using the DESeq2, EBSeq, SAMSeq, and NOISeq packages to select significantly differentially expressed genes between groups. 
The Combined method then claims a gene is significant only when it was claimed significant by all four methods. The following code showed 
an example of using the Comined_method function to get the list of differentially expressed genes between groups.

%code 3
<<eval=TRUE, echo=TRUE>>=
system.file("data", package = "ComSeq")
data(Bottomly_count)
summary(Bottomly_count)
label <- c(rep(0, 10), rep(1, 11))
Result <- Combined_method(RNAseqcount = Bottomly_count, label = label, alpha = 0.05)
summary(Result)
Result
@


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%{\bf Note: Sweave.} This document was generated using the \Rfunction{Sweave}
%function from the R \Rpackage{tools} package. The source file is in the
%\Rfunction{/inst/doc} directory of the package \Rpackage{marray}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\end{document}

